apps/src/netsim/NetSimPacketEditor.js

```diff
@@ -0,0 +1,495 @@
+/* jshint
+ funcscope: true,
+ newcap: true,
+ nonew: true,
+ shadow: false,
+ unused: true,
+
+ maxlen: 90,
+ maxparams: 3,
+ maxstatements: 200
+ */
+/* global $ */
+'use strict';
+
+require('../utils'); // For Function.prototype.inherits()
+var netsimMsg = require('../../locale/current/netsim');
+var markup = require('./NetSimPacketEditor.html');
+var KeyCodes = require('../constants').KeyCodes;
+var NetSimEncodingControl = require('./NetSimEncodingControl');
+var PacketEncoder = require('./PacketEncoder');
+var dataConverters = require('./dataConverters');
+var BITS_PER_BYTE = require('./netsimConstants').BITS_PER_BYTE;
+
+var minifyBinary = dataConverters.minifyBinary;
+var formatBinary = dataConverters.formatBinary;
+var formatHex = dataConverters.formatHex;
+var alignDecimal = dataConverters.alignDecimal;
+var binaryToInt = dataConverters.binaryToInt;
+var intToBinary = dataConverters.intToBinary;
+var hexToInt = dataConverters.hexToInt;
+var intToHex = dataConverters.intToHex;
+var hexToBinary = dataConverters.hexToBinary;
+var binaryToHex = dataConverters.binaryToHex;
+var decimalToBinary = dataConverters.decimalToBinary;
+var binaryToDecimal = dataConverters.binaryToDecimal;
+var asciiToBinary = dataConverters.asciiToBinary;
+var binaryToAscii = dataConverters.binaryToAscii;
+
+/**
+ * Generator and controller for message sending view.
+ * @param {Object} initialConfig
+ * @param {number} [initialConfig.toAddress]
+ * @param {number} [initialConfig.fromAddress]
+ * @param {number} [initialConfig.packetIndex]
+ * @param {number} [initialConfig.packetCount]
+ * @param {string} [initialConfig.message]
+ * @param {number} [initialConfig.maxPacketSize]
+ * @param {number} [initialConfig.chunkSize]
+ * @param {EncodingType[]} [initialConfig.enabledEncodings]
+ * @param {function} initialConfig.removePacketCallback
+ * @constructor
+ */
+var NetSimPacketEditor = module.exports = function (initialConfig) {
+
+  /**
+   * @type {jQuery}
+   * @private
+   */
+  this.rootDiv_ = $('<div>').addClass('netsim-packet');
+
+  /** @type {number} */
+  this.toAddress = initialConfig.toAddress || 0;
+  
+  /** @type {number} */
+  this.fromAddress = initialConfig.fromAddress || 0;
+  
+  /** @type {number} */
+  this.packetIndex = initialConfig.packetIndex || 1;
```

Comment:
Hm... right now they start at 1, but you're correct that I could future-proof this with a more explicit check.